### ISSUE TITLE
AnnotationRule false positive for multiple annotations with comment

### ISSUE THREAD

**Issue by kvn-stgl** [pinterest/ktlint#1539]
## Expected Behavior

We should be allowed to write multiple annotations on separate lines which are followed by comments. As example:
```
@Suppress("AnnotationRule") // some comment
@Suppress("AnnotationRule")
@Suppress("AnnotationRule")
class FooBar1 
```

Or for kotlin script:
```kotlin
@file:Suppress("AnnotationRule") // some comment
@file:DependsOn("ktlint")
@file:DependsOn("ktlint")
```

## Observed Behavior
The AnnotationRule is showing the following error: `Annotations with parameters should all be placed on separate lines prior to the annotated construct`

## Steps to Reproduce

The following test should show the problem:
```kotlin
    @Test
    fun `Given multiple annotations with a parameter followed by a EOL comment`() {
        val code =
            """
            @Suppress("AnnotationRule") // some comment
            @Suppress("AnnotationRule")
            @Suppress("AnnotationRule")
            class FooBar1
            """.trimIndent()

        annotationRuleAssertThat(code).hasNoLintViolations()
    }
```

## Your Environment

* Version of ktlint used: 0.46.1
